nx-X11/programs/Xserver/os/connection.c

```diff
@@ -988,7 +988,7 @@ CloseDownConnection(ClientPtr client)
     OsCommPtr oc = (OsCommPtr)client->osPrivate;
 
     if (FlushCallback)
-	CallCallbacks(&FlushCallback, NULL);
+	CallCallbacks(&FlushCallback, client);
```

Comment:
There actually was another issue with this commit. I backported the above commit from 2016 and then X.org commit c65f610e12f9df168d5639534ed3c2bd40afffc8 from 2010. I re-did this commit and the 2010 commit and wove them all into this one with extra backport commit message.